CesiumGltf/include/CesiumGltf/PropertyView.h

```diff
@@ -0,0 +1,2375 @@
+#include "CesiumGltf/ClassProperty.h"
+#include "CesiumGltf/PropertyTableProperty.h"
+#include "CesiumGltf/PropertyTextureProperty.h"
+#include "CesiumGltf/PropertyTypeTraits.h"
+
+#include <bitset>
+#include <cstring>
+#include <optional>
+
+namespace CesiumGltf {
+
+typedef int32_t PropertyViewStatusType;
+
+/**
+ * @brief Indicates the status of a property view.
+ *
+ * The {@link PropertyView} constructor always completes successfully.
+ * However, there may be fundamental errors with the property definition. In
+ * such cases, this enumeration provides the reason.
+ *
+ * This is defined with a class of static consts as opposed to an enum, so that
+ * derived property view classes can extend the statuses with their own specific
+ * errors.
+ */
+class PropertyViewStatus {
+public:
+  /**
+   * @brief This property view is valid and ready to use.
+   */
+  static const PropertyViewStatusType Valid = 0;
+
+  /**
+   * @brief This property view is trying to view a property that does not
+   * exist.
+   */
+  static const PropertyViewStatusType ErrorNonexistentProperty = 1;
+
+  /**
+   * @brief This property view's type does not match what is
+   * specified in {@link ClassProperty::type}.
+   */
+  static const PropertyViewStatusType ErrorTypeMismatch = 2;
+
+  /**
+   * @brief This property view's component type does not match what
+   * is specified in {@link ClassProperty::componentType}.
+   */
+  static const PropertyViewStatusType ErrorComponentTypeMismatch = 3;
+
+  /**
+   * @brief This property view differs from what is specified in
+   * {@link ClassProperty::array}.
+   */
+  static const PropertyViewStatusType ErrorArrayTypeMismatch = 4;
+
+  /**
+   * @brief This property says it is normalized, but is not of an integer
+   * component type.
+   */
+  static const PropertyViewStatusType ErrorInvalidNormalization = 5;
+
+  /**
+   * @brief The property provided an invalid offset value.
+   */
+  static const PropertyViewStatusType ErrorInvalidOffset = 6;
+
+  /**
+   * @brief The property provided an invalid scale value.
+   */
+  static const PropertyViewStatusType ErrorInvalidScale = 7;
+
+  /**
+   * @brief The property provided an invalid maximum value.
+   */
+  static const PropertyViewStatusType ErrorInvalidMax = 8;
+
+  /**
+   * @brief The property provided an invalid minimum value.
+   */
+  static const PropertyViewStatusType ErrorInvalidMin = 9;
+
+  /**
+   * @brief The property provided an invalid "no data" value.
+   */
+  static const PropertyViewStatusType ErrorInvalidNoDataValue = 10;
+
+  /**
+   * @brief The property provided an invalid default value.
+   */
+  static const PropertyViewStatusType ErrorInvalidDefaultValue = 11;
+};
+
+namespace {
```

Comment:
Ah, when I did that I was trying to convey that these helper functions are intended solely for this `.h` file, but I didn't realize it was bad form. I'll remove the namespace.

Reply:
I actually keep running into linking errors when I remove the anonymous namespace (for some function). This is the sort of thing that happens:
![image](https://github.com/CesiumGS/cesium-native/assets/32226860/db35fcfb-a805-4f9c-8e1e-8691da64adac)

It seems like the templated functions are fine, but any non-templated ones result in multiple definitions in the file. I'm not sure why this is happening... in any case, I'm keeping the anonymous namespace for the problematic functions, unless there is a better method of preventing this error

Reply:
on @csciguy8's suggestion, I'll append `inline` to the functions that are causing these linker errors. I think they're small enough to warrant the `inline` anyways

Reply:
Yeah you can either declare them inline, or you can move them to a .cpp file. Generally the latter is better because doing that helps keep compile times lower (not that it'll make a big difference in this case). Declaring them inline mostly only makes sense if you want them inlined for performance reasons.

Reply:
Maybe something's terribly wrong with my Visual Studio 2022 install, but moving them to the .cpp file keeps resolving in "unresolved external symbol" errors :/

Not sure what's wrong, this works just fine for `PropertyType.h`, which contains functions whose bodies are implemented in a `.cpp` file. Maybe it's got to do with the templating?

Reply:
Ah yep, sorry, I didn't realize those are templates. Templatize functions can't be defined in a .cpp file like that.

But also, including the complete definition in a header should _not_ result in multiply-defined symbols for template functions. I think the reason it is in your case is because you've declared some of the functions `static`. Remove that and it will work fine (I think).